### DISCUSSION TITLE
Use JS library in WASM

### DISCUSSION THREAD

**Discussion by TonyK-FmGroup** [unoplatform/uno#8685] [Q&A]
I am trying to use SignalR from a WASM project but am struggling to consume the JS library. Having started with a simple asp.net version, I know the code works in there so I'm using that as my baseline.

* WASM project with the signalR js files copied into WasmScripts/signalr/dist/browser
* This line in the csproj  `<EmbeddedResource Include="WasmScripts\signalr\dist\browser\signalr.js" />`
* An additional startup.js also in WasmScripts and in the csproj which executes when the WASM loads

This is the browser console output
![image](https://user-images.githubusercontent.com/103487880/166683457-1dcffccb-fdba-4f74-b315-c5945bc261ab.png)

Code is here:  [8685-UnoSignalR-main.zip](https://github.com/unoplatform/uno/files/8621738/8685-UnoSignalR-main.zip)

It feels as though the code is unable to locate the signalR library. I've also tried delaying the setup by executing the javascript only on a button press, in case it was trying to run before the library had been loaded, but the error is the same.

EDIT:
using dotnet 6, Uno 4.2.6,  VS 2022 preview 6 (also tried on preview 5 before updating today)

**Comment by jeromelaban** [comment]
SignalR is an AMD compatible dependency, which means you cannot access it directly from a script. To be able to use it you'll need to do the following:
```js
require([`${config.uno_app_base}/signalr.dist.browser.signalr`], signalR => {
    var connection = new signalR.HubConnectionBuilder().withUrl("/chatHub").build();
});
```

Note that your sample is including a folder structure in the WasmScript folder, and this is getting translated to a "dotted" path.

More details for the dependency management in Uno WebAssembly: https://github.com/unoplatform/Uno.Wasm.Bootstrap#dependency-management
